### PR TITLE
Add deprecation notice to documentation of the `aiven_service_user` r…

### PR DESCRIPTION
**Pull request by sharebear**
…esource

Today during a code review I noticed someone using this resource. When trying to find a reference of the deprecation it is only mentioned in the release notes and in the runtime deprecation warning, I would expect this to be present in the documentation of the resource too.

Adding a warning box for this, copying most of the phrasing from the runtime deprecation warning.

## About this change—what it does

Documentation improvement

## Why this way

I expect deprecated resources to be flagged as such in the provider documentation.

### CODE REVIEW

**Review by ivan-savciuc — DISMISSED** [review]
LGTM

**Comment by ivan-savciuc** [comment]
@sharebear pls run `make docs` and we can merge this

File: docs/resources/service_user.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ description: |-
 
 The Service User resource allows the creation and management of Aiven Service Users.
 
+~> **Note:** This resource is deprecated. Please use service-specific resources instead of this one, for example: aiven_kafka_user, aiven_pg_user etc.
```
Comment (Serpentiel) [inline docs/resources/service_user.md:13]: isn't this is supposed to be autogenerated?

Comment (ivan-savciuc) [reply docs/resources/service_user.md:13]: Yes indeed we have a resource deprecation message for `aiven_service_user` but it is not added to the generated docs.

Comment (ivan-savciuc) [reply docs/resources/service_user.md:13]: I think it has something to do with the docs gen tool

**Comment by sharebear** [comment]
Sorry for the slow response, I assumed the update was as simple as I regularly do with the azurerm provider. I'll get back to this tonight or tomorrow to give it a closer look.

**Comment by Serpentiel** [comment]
no problem, although please note, that it appears that it would require extra investigation since the docs generator [tries to remove this line](https://github.com/aiven/terraform-provider-aiven/actions/runs/3110833353/jobs/5042432542)

**Comment by ivan-savciuc** [comment]
@sharebear do you have time to run `make docs` and include those changes to the PR?

**Comment by sharebear** [comment]
> @sharebear do you have time to run `make docs` and include those changes to the PR?

I was intending to get around to this tonight but suddenly had a little incident at work that had to be dealt with. I'll mark this PR as draft for now.

**Comment by sharebear** [comment]
> no problem, although, please note, that it appears that it would require extra investigation since the docs generator [tries to remove this line](https://github.com/aiven/terraform-provider-aiven/actions/runs/3110833353/jobs/5042432542)

Yep, and according to this issue https://github.com/hashicorp/terraform-plugin-docs/issues/10 there's no possibility of using the deprecation notice from the schema. Therefore I went with the strategy of overriding the documentation template for the resource as has been done for a couple of other resources already.

**Review by ivan-savciuc — APPROVED** [review]
LGTM

**Pull request merged**